src/detect-parse.c

```diff
@@ -1243,6 +1243,16 @@ static int SigParse(DetectEngineCtx *de_ctx, Signature *s,
         memset(input, 0x00, buffer_size);
         memcpy(input, parser->opts, strlen(parser->opts)+1);
 
+        /* Make sure that options have a "sid" field at all.
```

Comment:
Suggest this check occurs after the `if` stmt on line 1239. Fail early before preparing `input` (zero-fill, then copy).

Also suggest error message be adjusted ... "Signature missing required value \"sid\""